src/main/drivers/vtx_rtc6705_soft_spi.c

```diff
@@ -49,6 +50,22 @@
 #define DISABLE_RTC6705       IOHi(rtc6705CsnPin)
 #define ENABLE_RTC6705        IOLo(rtc6705CsnPin)
 
+#if defined(USE_RTC6705_SOFTSPI_ON_HW_SPI)
```

Comment:
If this target requires changes outside of the target definition, then these should be submitted in a pull request separate from the new target, and will go into 4.0.

Also, there should be no target specific defines (like `USE_RTC6705_SOFTSPI_ON_HW_SPI`) as these changes will have to be done at runtime in 4.0. Please rewrite your driver changes to be applicable at runtime.

Reply:
OK, that's reasonable. 
I've seen codes in this way in many target, but sure it's bad for maintaining. 
Like `USE_RTC6705_CLK_HACK`, and etc. 

Then would you pls suggest any method i can use to do that.

Reply:
What the heck is this code doing? Disabling the hardware SPI function on these pins? Are these also used as SPI to access any other devices on this bus???

Reply:
@jflyper 
Yes, in the target we use, max7456 and rtc6705 share the same spi bus. But hardware spi is not working for rtc6705, that's the tradeoff.

Reply:
@czchc 
We are trying to make a generic target, in which a single firmware image is flashed onto all boards with the same MCU architecture. Can you tell us how your board fit into this scheme?

Reply:
@jflyper 
Okay, that sound awesome and new target and driver should not use the old way. I will check other drivers and find a generic way for this.